utils/src/test/scala/com/salesforce/op/utils/stats/StreamingHistogramTest.scala

```diff
@@ -0,0 +1,262 @@
+    cdfTest("Beta(1, 5)", beta2, grid, sampleSize, 100, 50)
+    cdfTest("Beta(1, 5)", beta2, grid, sampleSize, 200, 50)
+    cdfTest("Beta(1, 5)", beta2, grid, sampleSize, 400, 50)
+    cdfTest("0.5 * Beta(5, 1) + 0.5 * Beta(1, 5)", mixture, grid, sampleSize, 100, 50)
+    cdfTest("0.5 * Beta(5, 1) + 0.5 * Beta(1, 5)", mixture, grid, sampleSize, 200, 50)
+    cdfTest("0.5 * Beta(5, 1) + 0.5 * Beta(1, 5)", mixture, grid, sampleSize, 400, 50)
+  }
+
+  private def cdfTest(
+    distributionName: String,
+    dist: Distribution[Double] with HasCdf,
+    grid: Array[Double],
+    sampleSize: Int,
+    maxBins: Int,
+    numResults: Int): Unit = {
+
+    def getResults(sampleSize: Int, maxBins: Int): Array[DistributionTestResult] = {
+      val sample = (0 until sampleSize).map(_ => dist.draw()).toArray
+      val hist = new StreamingHistogram(maxBins)
+      val equiDist = equiDistBins(sample, maxBins)
+
+      hist.update(sample: _*)
+
+      grid.map { gridPoint =>
+        DistributionTestResult(
+          point = gridPoint,
+          trueCDF = dist.cdf(gridPoint),
+          streamingSumCDF = hist.sumCDF(gridPoint),
+          streamingEmpiricalCDF = hist.empiricalCDF(gridPoint),
+          equiDistSumCDF = StreamingHistogram.sumCDF(equiDist, gridPoint),
+          equiDistEmpiricalCDF = StreamingHistogram.empiricalCDF(equiDist, gridPoint))
+      }
+    }
+
+    def cdfMSE(allResults: Array[Array[DistributionTestResult]]): Array[MSEResult] = {
+      val points = allResults.flatMap(_.map(_.point)).distinct.sorted
+
+      def computeMSE(results: Array[DistributionTestResult], f: DistributionTestResult => Double): Double = {
+        val bias = breeze.stats.mean(results.map(r => r.trueCDF - f(r)))
+        val variance = breeze.stats.variance(results.map(f))
+
+        math.pow(bias, 2) + variance
+      }
+
+      points.map { pt =>
+        val results = allResults.map(_.filter(_.point == pt)(0))
+
+        MSEResult(
+          streamingSumCDFMSE = computeMSE(results, _.streamingSumCDF),
+          streamingEmpiricalCDFMSE = computeMSE(results, _.streamingEmpiricalCDF),
+          equiDistSumCDFMSE = computeMSE(results, _.equiDistSumCDF),
+          equiDistEmpiricalCDFMSE = computeMSE(results, _.equiDistEmpiricalCDF))
+      }
+    }
+
+    val mses = cdfMSE((0 until numResults).map(_ => getResults(sampleSize, maxBins)).toArray)
+
+    val MeanAndVariance(streamingSumCDFMSEMean, streamingSumCDFMSEVar, _) =
+      meanAndVariance(mses.map(_.streamingSumCDFMSE))
+    val MeanAndVariance(streamingEmpiricalCDFMSEMean, streamingEmpiricalCDFMSEVar, _) =
+      meanAndVariance(mses.map(_.streamingEmpiricalCDFMSE))
+    val MeanAndVariance(equiDistSumCDFMSEMean, equiDistSumCDFMSEVar, _) =
+      meanAndVariance(mses.map(_.equiDistSumCDFMSE))
+    val MeanAndVariance(equiDistEmpiricalCDFMSEMean, equiDistEmpiricalCDFMSEVar, _) =
+      meanAndVariance(mses.map(_.equiDistEmpiricalCDFMSE))
+
+    println("-" * 50)
```

Comment:
extract printing into a helper function in the spec + replace println with log.info